pkg/detector/detector.go

```diff
@@ -1401,15 +1412,10 @@ func (d *ResourceDetector) CleanupResourceBindingLabels(rb *workv1alpha2.Resourc
 	})
 }
 
-// CleanupClusterResourceBindingLabels removes labels from cluster resource binding.
-func (d *ResourceDetector) CleanupClusterResourceBindingLabels(crb *workv1alpha2.ClusterResourceBinding, labels ...string) error {
-	bindingLabels := crb.GetLabels()
-	for _, l := range labels {
-		delete(bindingLabels, l)
-	}
-
+// cleanupCRBLabels removes labels from cluster resource binding.
+func (d *ResourceDetector) cleanupCRBLabels(crb *workv1alpha2.ClusterResourceBinding, removeLabels RemoveMapKeysFunc) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		crb.SetLabels(bindingLabels)
+		crb.SetLabels(removeLabels(crb.GetLabels()))
 		updateErr := d.Client.Update(context.TODO(), crb)
```

Comment:
How about:
```suggestions
	bindingLabels := removeLabels(crb.GetLabels())
	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
		crb.SetLabels(bindingLabels)
```
Avoid repeatedly calling `removeLabels` when retrying to prevent failure.